apps/frontend/src/libs/components/date-input/date-input.tsx

```diff
@@ -0,0 +1,119 @@
+import { Calendar } from "react-calendar";
+import {
+	type Control,
+	type FieldPath,
+	type FieldValues,
+} from "react-hook-form";
+
+import { Icon, Popover } from "~/libs/components/components.js";
+import {
+	addDays,
+	formatDate,
+	getDifferenceInDays,
+} from "~/libs/helpers/helpers.js";
+import {
+	useCallback,
+	useFormController,
+	usePopover,
+} from "~/libs/hooks/hooks.js";
+
+import { type DateValue } from "./libs/types/types.js";
+import styles from "./styles.module.css";
+
+type Properties<T extends FieldValues> = {
+	control: Control<T, null>;
+	maxDate?: Date;
+	maxRange?: number;
+	name: FieldPath<T>;
+};
+
+const DateInput = <T extends FieldValues>({
+	control,
+	maxDate,
+	maxRange,
+	name,
+}: Properties<T>): JSX.Element => {
+	const { field } = useFormController({
+		control,
+		name,
+	});
+
+	const { isOpened, onClose, onOpen } = usePopover();
+
+	const getSelectedDateRange = (): string => {
+		const [startDate, endDate] = field.value as Date[];
+
+		if (!startDate || !endDate) {
+			return "Choose date range";
+		}
+
+		return `${formatDate(startDate, "MMM d, yyyy")} - ${formatDate(endDate, "MMM d, yyyy")}`;
+	};
+
+	const handleDateChange = useCallback(
+		(dates: DateValue): void => {
+			const [startDate, endDate] = dates as Date[];
+
+			if (maxRange && startDate && endDate) {
+				const diffInDays = getDifferenceInDays(endDate, startDate);
+
+				if (diffInDays > maxRange) {
+					const adjustedEndDate = addDays(startDate, maxRange);
+					field.onChange([startDate, adjustedEndDate]);
+
+					return;
+				}
+			}
+
+			field.onChange(dates);
+		},
+		[field, maxRange],
+	);
+
+	const handleFormatShortWeekday = useCallback(
+		(_locale: string | undefined, date: Date): string => {
+			return formatDate(date, "EEEEEE");
+		},
+		[],
+	);
+
+	return (
+		<div>
+			<Popover
+				content={
+					<Calendar
+						calendarType="gregory"
+						className={styles["calendar"]}
+						formatShortWeekday={handleFormatShortWeekday}
+						locale="en"
+						maxDate={maxDate ?? new Date()}
```

Comment:
By default there shouldn't be any limitation for the future date. we need to pass this through props and include this property here, only if it's passed

Reply:
Changed everything. Thanks!